src/main/java/org/openshift/jenkins/plugins/openshiftlogin/OpenShiftOAuth2SecurityRealm.java

```diff
@@ -232,19 +305,25 @@ public Authentication authenticate(Authentication authentication) throws Authent
     }
 
     protected OAuthSession newOAuthSession(String from, final String redirectOnFinish) {
-    	final GenericUrl tokenServerURL = new GenericUrl(serverPrefix + "/oauth/token");
-        final String authorizationServerURL = serverPrefix +"/oauth/authorize";
+    	LOGGER.fine(String.format("newOAuthSession from %s redirectOnFinish %s", from, redirectOnFinish));
+    	final GenericUrl tokenServerURL = new GenericUrl(getServerPrefix(overrides) + "/oauth/token");
+        final String authorizationServerURL = getServerPrefix(overrides) +"/oauth/authorize";
+        
+        LOGGER.fine(String.format("tokenServerURL %s auth server url %s clientid %s clientSecret %s transport %s", tokenServerURL, authorizationServerURL, getClientId(overrides), getClientSecret(overrides) == null ? null : getClientSecret(overrides).getPlainText(), transport == null ? null : transport.toString()));
         
         final AuthorizationCodeFlow flow = new AuthorizationCodeFlow.Builder(
                 BearerToken.queryParameterAccessMethod(), transport, JSON_FACTORY, tokenServerURL,
-                new ClientParametersAuthentication(clientId, clientSecret.getPlainText()), clientId, authorizationServerURL)
+                new ClientParametersAuthentication(getClientId(overrides), getClientSecret(overrides).getPlainText()), getClientId(overrides), authorizationServerURL)
```

Comment:
Is this a common pattern in Jenkins?  It seems a violation of the object encapsulation pattern, and seems arbitrary.  Why not `getDefaultedClientId` instead and make it strongly typed?